bliss/encoder/encoder.py

```diff
@@ -36,6 +37,7 @@ def __init__(
         sample_image_renders: MetricCollection,
         matcher: CatalogMatcher,
         min_flux_threshold: float = 0,
+        min_flux_threshold_during_test: float = 0,
```

Comment:
Did you mean to change this variable name?

Reply:
`min_flux_threshold` is used during training. `min_flux_threshold_during_test` is used to filter true_cat and our prediction during test. Now, we set both of them to 100.

Reply:
I see. It's so unusual to have different thresholds during training and validation/testing. With different thresholds, you can't even compare training loss and validation loss to get an idea of how much you're overfitting, right?

Reply:
Could we filter the catalogs when computing validation and testing metrics, rather than when we compute loss?

Reply:
yes. i design this flag according to our previous discussion. we may face the case where we need to perform training using data with flux over 90 and test the model using data of higher flux (for example, 100). if we don't need to test this case in future, we can remove it from this PR.

Reply:
> Could we filter the catalogs when computing validation and testing metrics, rather than when we compute loss?

do you mean we need to train the model with unfiltered data?

Reply:
I think we should use the same threshold for computing training loss and validation/testing loss (e.g., 90). But use a potentially different threshold for computing the validation metrics (e.g., 100). Does that make sense? The threshold for metrics could be an argument passed to the metric even rather than an argument of the encoder.

Reply:
If you set `exclude_last_bin` to True for `DetectionMetrics`, and the last bin cutoff is 100 (converted to magnitude), then you're effectively doing this.

Reply:
yes, this is what we did in our current code, except that we filter the objects in encoder. Conducting this step in encoder may simplify the design logic. if we pass this argument to metric, we may need to modify the interface and code for each metric class.

Reply:
OK I get it now :-)